plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTableProperties.java

```diff
@@ -128,4 +158,31 @@ public static ColumnMappingMode getColumnMappingMode(Map<String, Object> tablePr
     {
         return ColumnMappingMode.valueOf(tableProperties.get(COLUMN_MAPPING_MODE_PROPERTY).toString().toUpperCase(ENGLISH));
     }
+
+    public static Map<String, String> getExtraProperties(Map<String, Object> tableProperties)
+    {
+        Map<String, String> extraProperties = (Map<String, String>) tableProperties.get(EXTRA_PROPERTIES);
+        if (extraProperties != null) {
+            return extraProperties;
+        }
+        return emptyMap();
+    }
+
+    private static void validateExtraProperties(Map<String, String> extraProperties)
+    {
+        if (extraProperties.containsValue(null)) {
+            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Extra table property value cannot be null '%s'", extraProperties));
+        }
+        if (extraProperties.containsKey(null)) {
+            throw new TrinoException(INVALID_TABLE_PROPERTY, format("Extra table property key cannot be null '%s'", extraProperties));
+        }
+        Set<String> illegalExtraProperties = Sets.intersection(
+                DeltaLakeTableProperties.PROPERTY_NAMES,
```

Comment:
In the Hive connector we disallow using the key `extra_properties` in the `extra_properties` https://github.com/trinodb/trino/blob/9d95bfff1ebf5788f4ac162f4480ac17748f25d3/plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java#L1236

Disallowing setting table properties through `extra_properties` and through ordinary properties reduces some possible user confusion, but also limits the use of `extra_properties`. For Hive this could make sense, as Trino table property names might overlap with Hive table property names with special meanings. For Delta it might not matter, as none of our current table properties have a special meaning as a Delta table property (special Delta table properties all seem to be prefixed with `delta.`. Perhaps we should get rid of this validation of illegal `extra_property` keys, and only check for the special `delta.` keys?